nucliadb/nucliadb/common/context/__init__.py

```diff
@@ -84,11 +84,11 @@ async def finalize(self) -> None:
         if not self._initialized:
             return
 
+        await stop_nats_manager()
```

Comment:
nit: in theory, the tear down sequence should be in the reverse order of the creation

Reply:
totally agree! otherwise we can have unexpected exceptions on service teardowns!
Changing